src/_P075_Nextion.ino

```diff
@@ -1,308 +1,620 @@
+            Settings.BaudRate = AdvHwBaud;              // Set BaudRate for Nextion.
+            Serial.flush();
+            Serial.begin(AdvHwBaud);
+            Serial.swap();
+        }
+        // Hardware serial is RX on 3 and TX on 1. USB serial for Nextion IDE (User MCU Input function).
+        else if(AdvHwSerial && rxPin == 3 && txPin == 1) {
+            log = F("NEXTION075 : Using USB hardware serial");
+            addLog(LOG_LEVEL_INFO, log);
+            HwSerial = UARTSERIAL;
+            Settings.UseSerial = false;                 // Disable global Serial port.
+            Settings.SerialLogLevel = 0;                // Disable logging on serial port.
+            Settings.BaudRate = AdvHwBaud;              // Set BaudRate for Nextion.
+            Serial.flush();
+            Serial.begin(AdvHwBaud);
+        }
+        else {
+            log = F("NEXTION075 : Using software serial");
+            addLog(LOG_LEVEL_INFO, log);
+            HwSerial = SOFTSERIAL;
+            if (SoftSerial == NULL) {
+                SoftSerial = new ESPeasySoftwareSerial(rxPin, txPin, false, RXBUFFSZ);
+            } 
+            SoftSerial->begin(9600);
+            SoftSerial->flush();
+        }
+    }
+    else {
+    }
       success = true;
       break;
     }
 
-    case PLUGIN_EXIT:
-      {
-          if (nextion)
-          {
-            delete nextion;
-            nextion=NULL;
-          }
-          break;
-      }
 
-    case PLUGIN_READ: {
-        char deviceTemplate[Nlines][64];
-        LoadCustomTaskSettings(event->TaskIndex, (byte*)&deviceTemplate, sizeof(deviceTemplate));
+    case PLUGIN_READ: {    
+        char deviceTemplate[Nlines][Lenlines];
+        int RssiIndex;
         String newString;
+        String tmpString;
+        String UcTmpString;
+
+        LoadCustomTaskSettings(event->TaskIndex, (byte*)&deviceTemplate, sizeof(deviceTemplate));
 
+// Get optional LINE command statements. Special RSSIBAR bargraph keyword is supported.
         for (byte x = 0; x < Nlines; x++) {
-          String tmpString = deviceTemplate[x];
-          if (tmpString.length())
-          {
-            int rssiIndex = tmpString.indexOf(F("rssi"));
-            if(rssiIndex >= 0)
-            {
+          tmpString = deviceTemplate[x];
+          if (tmpString.length()) {
+            UcTmpString = deviceTemplate[x];
+            UcTmpString.toUpperCase();
+            RssiIndex = UcTmpString.indexOf(F("RSSIBAR"));  // RSSI bargraph Keyword found, wifi value in dBm.
+            if(RssiIndex >= 0) {
               int barVal;
-              newString = tmpString.substring(0, rssiIndex);
+              newString = tmpString.substring(0, RssiIndex);
               int nbars = WiFi.RSSI();
-              if (nbars < -100)
+              if (nbars < -100 || nbars >= 0)
                  barVal=0;
-              else if (nbars >= -100 and nbars < -90)
+              else if (nbars >= -100 && nbars < -95)
```

Comment:
This big if-else statement looks a bit like 
```C++
barVal = 0;
if (nbars >= -100 && nbars < 0) {
  barVal = (nbars + 100) / 10; 
  barVal *= 10; 
  if (barVal == 0) barVal = 5;
}
```

Reply:
@TD-er:  It's not a straight linear conversion. The barVal is an extrapolation of the log based nbars (dBm). The stepped values have been chosen based on my experience with RF receiver sensitivity and typical expected RF range. The result in intended for a Nextion bar-graph gauge component that is scaled in a 0-100 data range. Can be used to create a signal bar graphic like is  seen on a cell phone.

- Thomas

Reply:
A switch statement would not have been easier, since then you would have to assign either an enum, or some character to each range and that would still have meant a lot of if-then-else statements.

Reply:
I forgot to say thank you for the approval / merge / commit. I hope I didn't break anything. :)
- Thomas

Reply:
I hope you did test it, since I didn't ;)